File: engine/__tests__/glossary.test.ts
import { lookupTerm, listGlossary } from '../glossary';

describe('glossary lookups', () => {
  test('finds known terms case-insensitively', () => {
    expect(lookupTerm('esrs')).not.toBeNull();
    expect(lookupTerm('ESRS')).not.toBeNull();
    expect(lookupTerm('Esrs')).not.toBeNull();
  });

  test('returns null for unknown terms', () => {
    expect(lookupTerm('not-a-real-term')).toBeNull();
    expect(lookupTerm('')).toBeNull();
  });

  test('common abbreviations are all registered', () => {
    for (const term of ['QTc', 'EPS', 'akathisia', 'MAOI', 'LAI', 'CYP2D6', 'FBC', 'eGFR']) {
      expect(lookupTerm(term)).not.toBeNull();
    }
  });

  test('listGlossary returns alphabetical list', () => {
    const list = listGlossary();
    expect(list.length).toBeGreaterThan(15);
    for (let i = 1; i < list.length; i++) {
      expect(list[i].term.localeCompare(list[i - 1].term)).toBeGreaterThanOrEqual(0);
    }
  });

  test('every entry has a non-empty definition', () => {
    for (const e of listGlossary()) {
      expect(e.definition.length).toBeGreaterThan(0);
    }
  });
});

File: flutter_app/test/engine/glossary_test.dart
// Tests for the Dart glossary port. Mirrors engine/__tests__/glossary.test.ts.

import 'package:flutter_test/flutter_test.dart';
import 'package:psychswitch/src/engine/glossary.dart';

void main() {
  group('glossary lookups', () {
    test('finds known terms case-insensitively', () {
      expect(lookupTerm('esrs'), isNotNull);
      expect(lookupTerm('ESRS'), isNotNull);
      expect(lookupTerm('Esrs'), isNotNull);
    });

    test('returns null for unknown terms', () {
      expect(lookupTerm('not-a-real-term'), isNull);
      expect(lookupTerm(''), isNull);
    });

    test('common abbreviations are all registered', () {
      const abbrevs = <String>[
        'QTc',
        'EPS',
        'akathisia',
        'MAOI',
        'LAI',
        'CYP2D6',
        'FBC',
        'eGFR',
      ];
      for (final term in abbrevs) {
        expect(lookupTerm(term), isNotNull, reason: 'missing: $term');
      }
    });

    test('listGlossary returns alphabetical list', () {
      final list = listGlossary();
      expect(list.length, greaterThan(15));
      for (var i = 1; i < list.length; i++) {
        expect(
          list[i].term.compareTo(list[i - 1].term),
          greaterThanOrEqualTo(0),
        );
      }
    });

    test('every entry has a non-empty definition', () {
      for (final e in listGlossary()) {
        expect(e.definition, isNotEmpty);
      }
    });

    test('lookup trims whitespace', () {
      expect(lookupTerm('  qtc  '), isNotNull);
    });

    test('relevance is preserved when present and absent when not', () {
      // 'esrs' has a relevance hint; 'ssri' does not.
      expect(lookupTerm('esrs')?.relevance, isNotNull);
      expect(lookupTerm('ssri')?.relevance, isNull);
    });
  });
}
